edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,71 @@
 """ Permission classes. """
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.exceptions import PermissionDenied
 from rest_framework.permissions import BasePermission
 
+from edx_rest_framework_extensions.decorators import (
+    jwt_authentication_required,
+    skip_unless_jwt_scopes_enforced,
+)
+from edx_rest_framework_extensions.utils import jwt_decode_handler
+
 
 class IsSuperuser(BasePermission):
     """ Allows access only to superusers. """
 
     def has_permission(self, request, view):
         return request.user and request.user.is_superuser
+
+
+class JwtHasScope(TokenHasScope):
+    """
+    The request is authenticated as a user and the token used has the right scope.
+    """
+
+    @skip_unless_jwt_scopes_enforced
+    @jwt_authentication_required
+    def has_permission(self, request, view):
+        jwt_scopes = jwt_decode_handler(request.auth).get('scopes', [])
```

Comment:
For separation of concerns, does it make sense for all JWT token parsing to live in one python module?  So move the accessing of `scopes` from the token near `jwt_decode_handler`?

Reply:
Meaning, add a new function to `edx_rest_framework_extensions.utils` that accepts the encoded JWT and returns the list of scopes?